bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -262,6 +258,32 @@ public void partClosed(IWorkbenchPart part) {
 		public void partOpened(IWorkbenchPart part) {
 			// Do nothing
 		}
+
+		private void adaptToPartActivationChange() {
+			if (getShell() == null || targetPart.getSite().getShell() == null) {
+				return;
+			}
+
+			getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+
+			targetPart.getSite().getShell().setActive();
+			targetPart.getSite().getPart().setFocus();
+
+			Display.getDefault().asyncExec(this::focusTargetWidget);
+		}
+
+		private void focusTargetWidget() {
+			if (targetPart.getSite().getShell() == null) {
```

Comment:
`targetPart.getSite().getShell() == null` should be replaced with `getShell() == null || targetPart.getSite().getPart() == null`

Reply:
I don't quite understand why that makes sense, but if it is true that `targetPart.getSite().getPart() == null` implies `targetPart.getSite().getShell() == null`, then that should work.

Please keep in mind that this code is ran from inside a Dialog, so `getShell()` does not return the same shell as the shell the part is located in.

Reply:
I performed the changes in https://github.com/eclipse-platform/eclipse.platform.ui/commit/ad63e43557ee5c01ce8857d21d4dc5436905f9bb